ansible_base/oauth2_provider/views/permissions.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import permissions
+
+
+class OAuth2TokenPermission(permissions.BasePermission):
+    # An app token is a token that has an application attached to it
+    # A personal access token (PAT) is a token with no application attached to it
+    # With that in mind:
+    # - An app token can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the admin of the organization of the application of the token
+    #   - I am the user of the token
+    # - An app token can be created if:
+    #   - I have read access to the application (currently this means: I am the superuser)
+    # - A PAT can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the user of the token
+    # - A PAT can be created if:
+    #   - I am a user
+
+    def has_permission(self, request, view):
+        # Handle PAT and app token creation separately
+        if request.method == 'POST':
+            if request.data.get('application'):
+                # TODO: Change this once dab#424 is fixed
+                return request.user.is_superuser
```

Comment:
:+1: bugs processing `request.data` are a nagging concern of mine, but this is simple enough that it'll be perfectly fine for now.